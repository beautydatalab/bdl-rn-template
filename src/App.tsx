/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * Generated with the TypeScript template
 * https://github.com/react-native-community/react-native-template-typescript
 *
 * @format
 */

import {NavigationContainer} from '@react-navigation/native';
import React, {Component} from 'react';
import {StatusBar} from 'react-native';
import SplashScreen from 'react-native-splash-screen';
import MainBottomTabNavigator from 'src/navigators/MainBottomTabNavigator';

export default class App extends Component {
  public render() {
    return (
      <NavigationContainer>
        <StatusBar barStyle="dark-content" />
        <MainBottomTabNavigator />
      </NavigationContainer>
    );
  }

  public componentDidMount() {
    SplashScreen.hide();
  }
}
