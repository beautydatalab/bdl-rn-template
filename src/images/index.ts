export default {
  header: require('./header-image.jpeg'),
};
