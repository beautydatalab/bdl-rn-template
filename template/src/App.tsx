/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * Generated with the TypeScript template
 * https://github.com/react-native-community/react-native-template-typescript
 *
 * @format
 */

import React, {Component} from 'react';

import {StatusBar} from 'react-native';

import {NavigationContainer} from '@react-navigation/native';
import {Provider} from 'mobx-react';
import SplashScreen from 'react-native-splash-screen';

import {MainBottomTabNavigator} from 'src/navigators/MainBottomTabNavigator';
import {getRootStore} from 'src/stores/RootStore';

export default class App extends Component {
  public render() {
    return (
      <Provider store={getRootStore}>
        <NavigationContainer>
          <StatusBar barStyle="dark-content" />
          <MainBottomTabNavigator />
        </NavigationContainer>
      </Provider>
    );
  }

  public componentDidMount() {
    SplashScreen.hide();
  }
}
