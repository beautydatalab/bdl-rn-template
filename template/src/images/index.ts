export const images = {
  header: require('./header-image.jpeg'),
};
