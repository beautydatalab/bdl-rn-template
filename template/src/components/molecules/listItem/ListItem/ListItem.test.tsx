import React from 'react';

import renderer from 'react-test-renderer';
import styled from 'styled-components/native';

import {images} from 'src/images';

import {ListItem} from './index';

const Image = styled.Image`
  width: 50px;
  height: 50px;
  border-radius: 25px;
`;

test('render correctly', () => {
  const tree = renderer
    .create(
      <>
        <ListItem>
          <Image source={images.header} />
          <ListItem.Content>
            <ListItem.Title>Image</ListItem.Title>
          </ListItem.Content>
        </ListItem>
        <ListItem chevron>
          <Image source={images.header} />
          <ListItem.Content>
            <ListItem.Title>Chevron</ListItem.Title>
          </ListItem.Content>
        </ListItem>
        <ListItem chevron bottomDivider>
          <Image source={images.header} />
          <ListItem.Content>
            <ListItem.Title>BottomDivider</ListItem.Title>
          </ListItem.Content>
        </ListItem>
        <ListItem chevron bottomDivider>
          <Image source={images.header} />
          <ListItem.Content>
            <ListItem.Title>Title</ListItem.Title>
            <ListItem.SubTitle>SubTitle</ListItem.SubTitle>
          </ListItem.Content>
        </ListItem>
      </>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
