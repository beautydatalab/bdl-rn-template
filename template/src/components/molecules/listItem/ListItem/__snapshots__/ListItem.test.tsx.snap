// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
Array [
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomWidth": 0,
            "borderColor": "#d5d5d5",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/images/header-image.jpeg",
          }
        }
        style={
          Array [
            Object {
              "borderRadius": 25,
              "height": 50,
              "width": 50,
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "flexBasis": 0,
              "flexDirection": "column",
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "paddingHorizontal": 8,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#212121",
              },
              Array [
                Object {},
              ],
              Object {
                "fontSize": 20,
              },
              Object {
                "color": "#212121",
              },
            ]
          }
        >
          Image
        </Text>
      </View>
    </View>
  </View>,
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomWidth": 0,
            "borderColor": "#d5d5d5",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/images/header-image.jpeg",
          }
        }
        style={
          Array [
            Object {
              "borderRadius": 25,
              "height": 50,
              "width": 50,
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "flexBasis": 0,
              "flexDirection": "column",
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "paddingHorizontal": 8,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#212121",
              },
              Array [
                Object {},
              ],
              Object {
                "fontSize": 20,
              },
              Object {
                "color": "#212121",
              },
            ]
          }
        >
          Chevron
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#212121",
            },
            Array [
              Object {},
            ],
            Object {
              "fontSize": 24,
            },
            Object {
              "color": "#d5d5d5",
            },
          ]
        }
      >
        &gt;
      </Text>
    </View>
  </View>,
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      bottonDivider={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#d5d5d5",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/images/header-image.jpeg",
          }
        }
        style={
          Array [
            Object {
              "borderRadius": 25,
              "height": 50,
              "width": 50,
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "flexBasis": 0,
              "flexDirection": "column",
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "paddingHorizontal": 8,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#212121",
              },
              Array [
                Object {},
              ],
              Object {
                "fontSize": 20,
              },
              Object {
                "color": "#212121",
              },
            ]
          }
        >
          BottomDivider
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#212121",
            },
            Array [
              Object {},
            ],
            Object {
              "fontSize": 24,
            },
            Object {
              "color": "#d5d5d5",
            },
          ]
        }
      >
        &gt;
      </Text>
    </View>
  </View>,
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      bottonDivider={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#d5d5d5",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/images/header-image.jpeg",
          }
        }
        style={
          Array [
            Object {
              "borderRadius": 25,
              "height": 50,
              "width": 50,
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "flexBasis": 0,
              "flexDirection": "column",
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "paddingHorizontal": 8,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#212121",
              },
              Array [
                Object {},
              ],
              Object {
                "fontSize": 20,
              },
              Object {
                "color": "#212121",
              },
            ]
          }
        >
          Title
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#212121",
              },
              Array [
                Object {},
              ],
              undefined,
              Object {
                "color": "#909090",
              },
            ]
          }
        >
          SubTitle
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#212121",
            },
            Array [
              Object {},
            ],
            Object {
              "fontSize": 24,
            },
            Object {
              "color": "#d5d5d5",
            },
          ]
        }
      >
        &gt;
      </Text>
    </View>
  </View>,
]
`;
